### PR TITLE
Migrate Travis CI to Github Actions

### PR DESCRIPTION
**Pull request by pnbabu**
- Fix build errors by modifying the code compatible with the latest nest master
- Migrate CI from Travis to Github Actions

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -29,12 +29,25 @@
 set -e
 
 # We need to do this, because  update-alternatives is not available on MacOS
-if [[ $OSTYPE = darwin* ]]; then
-  export CC=$(ls /usr/local/bin/gcc-* | grep '^/usr/local/bin/gcc-\d$')
-  export CXX=$(ls /usr/local/bin/g++-* | grep '^/usr/local/bin/g++-\d$')
+# if [[ $OSTYPE = darwin* ]]; then
+#   export CC=$(ls /usr/local/bin/gcc-* | grep '^/usr/local/bin/gcc-\d$')
+#   export CXX=$(ls /usr/local/bin/g++-* | grep '^/usr/local/bin/g++-\d$')
+# fi
```
Comment (jougs) [inline build.sh:35]:
```suggestion
```

File: build.sh [hunk 2]
```diff
@@ -51,17 +64,26 @@ echo "--> Detected PYTHON_INCLUDE_DIR=$PYTHON_INCLUDE_DIR"
 # Explicitly allow MPI oversubscription. This is required by Open MPI versions > 3.0.
 # Not having this in place leads to a "not enough slots available" error.
 cp extras/nestrc.sli ~/.nestrc
-if [[ "$OSTYPE" = "darwin"* ]] ; then
-  sed -i -e 's/mpirun -np/mpirun --oversubscribe -np/g' ~/.nestrc
+# if [[ "$OSTYPE" = "darwin"* ]] ; then
+#   sed -i -e 's/mpirun -np/mpirun --oversubscribe -np/g' ~/.nestrc
+# fi
```
Comment (jougs) [inline build.sh:69]:
```suggestion
```

**Review by jougs — APPROVED** [review]
Thanks! :smile_cat:

**Review by clinssen — APPROVED** [review]
Thanks!

**Pull request merged**